### DISCUSSION TITLE
Nested compound/stem variables.

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#361] [Ideas]
What I would like is some on-the-fly calculation within stems. The content within the brackets are calculated before addressing the stem: 

**mystem.[i+3].list**

with this method, we could also have stems within stems: 

**mystem.[this.stem.i].list** means the second stem's content is "inserted" before addressing the first stem.

and of course, we can mix both: 
**mystem.[this.stem.i*3+1].list**

**Comment by adesutherland** [comment]
At the moment we only have arrays but for stems and objects this is near the plan. Just a minor syntax clarification:

Using stem syntax we have

    j = i + 3
    say mystem.j.list

In alternative square bracket syntax it would be written as

    say mystem[i+3,list]
    say mystem[stem.i * 3 + 1, list]

You can't mix stem and square bracket syntax for **arrays** because arrays are multi-dimensional, so

    say mystem[1+3].list 

would give an error ARRAY_DIMS_MISMATCH as it is expecting 2 dimensions in the square brackets.

For objects (and therefore stems), it is more or less like you suggest.

    object.index

could return an object or an array and you could cascade with

    object.index.anotherindex.methodcall()

or 

    object[index].anotherindex.methodcall()

or 

    object[index][anotherindex].methodcall()

What I am not sure about, though, is whether this should work or not

    object[index, anotherindex].methodcall()

On balance, no, because this makes the embedding of objects unclear to the reader / coder

Clear as mud?
